### ISSUE TITLE
Use for Ham Radio operators?

### ISSUE THREAD

**Issue by wgaylord** [cnlohr/esp8266oddclock#2]
Just tossing this as an idea. Considering part of the Wi-Fi freq are able to be used by hams in the US.

You could use promiscuous mode to receive and raw frames to transmit.

Any idea about is this works with the ESP32?

**Comment by cnlohr** [comment]
You can totally do raw frames, in fact that's what I'm doing here. Raw frames FTW!

The only thing I wish I could do is figure out how the RFPLL works so I could do custom frequencies.

And re: ESP32: Sadly, it's a little more complicated to dig into so I haven't really gotten the chance to and haven't particularly had the need to.

**Comment by DL4APT** [comment]
I always dreamed of an ESP capable of direct use on the Hamnet (https://hamnetdb.net/map.cgi). In the short time I looked into the documentation and code I can assure that the RFPLL is indeed a hard nut to crack.

**Comment by wgaylord** [comment]
I think we need to try and get they to release information about their wifi stack, As the WiFi stack uses the 2.4 GHZ SDR to do Wifi on the ESP32 (Not sure about the ESP8266.)

**Comment by cnlohr** [comment]
I don't think that will work. Espressif holds all of this information _very_ tightly and they have a vested interest to keep their position in the market from being eaten up by the next smaller guys who will be nipping at their ankles.  As of now, Espressif's radios are top notch, for a while the ESP8266 outperformed even the most expensive radios you could get on any Cisco gear. 

Feel free to try to get the information, but I don't think you're going to get very far without a *VERY* compelling argument.  It would probably take millions of dollars worth of sales, and with razor thin margins, that could take quite some time.

**Comment by wgaylord** [comment]
cnlohr Well its not Espressif that is holding the information. Its the actually producer of the core. As Espressif has said they can't directly release the full stacks source code because its not their IP.

I wish they could at least document the cores SDR api. Even if its just low level information. As the documents for the ESP32 mention that it is an SDR.
